cns/requestcontroller/kubecontroller/crdtranslator.go

```diff
@@ -1,19 +1,80 @@
 package kubecontroller
 
 import (
+	"net"
+	"strconv"
+	"strings"
+
 	"github.com/Azure/azure-container-networking/cns"
 	nnc "github.com/Azure/azure-container-networking/nodenetworkconfig/api/v1alpha"
 )
 
-// CRDStatusToNCRequest translates a crd status to network container request
+// CRDStatusToNCRequest translates a crd status to createnetworkcontainer request
 func CRDStatusToNCRequest(crdStatus *nnc.NodeNetworkConfigStatus) (*cns.CreateNetworkContainerRequest, error) {
-	//TODO: Translate CRD status into network container request
-	//Mat will pick up from here
-	return nil, nil
+	var (
+		createNCRequest   cns.CreateNetworkContainerRequest
+		nc                nnc.NetworkContainer
+		secondaryIPConfig cns.SecondaryIPConfig
+		ipSubnet          cns.IPSubnet
+		ipAssignment      nnc.IPAssignment
+		prefixLength      int
+	)
+
+	createNCRequest.SecondaryIPConfigs = make(map[string]cns.SecondaryIPConfig)
+
+	for _, nc = range crdStatus.NetworkContainers {
+		createNCRequest.NetworkContainerid = nc.ID
+		createNCRequest.PrimaryInterfaceIdentifier = nc.PrimaryIP
+		prefixLength = maskStringToPrefixLength(nc.Netmask)
+
+		for _, ipAssignment = range nc.IPAssignments {
+			ipSubnet.IPAddress = ipAssignment.IP
+			ipSubnet.PrefixLength = uint8(prefixLength)
+			secondaryIPConfig = cns.SecondaryIPConfig{
+				IPConfig: ipSubnet,
+			}
+			createNCRequest.SecondaryIPConfigs[ipAssignment.Name] = secondaryIPConfig
+		}
+	}
+
+	//Only returning the first network container for now, later we will return a list
+	return &createNCRequest, nil
 }
 
 // CNSToCRDSpec translates CNS's list of Ips to be released and requested ip count into a CRD Spec
-func CNSToCRDSpec() (*nnc.NodeNetworkConfigSpec, error) {
-	//TODO: Translate list of ips to be released and requested ip count to CRD spec
-	return nil, nil
+func CNSToCRDSpec(secondaryIPConfigs []cns.SecondaryIPConfig, ipCount int) (*nnc.NodeNetworkConfigSpec, error) {
```

Comment:
Can we rename secondaryIPConfigs  to secondaryIpsNotInUse or tobeDeletedSecondaryIpConfigs